scripts/update-npm.sh

```diff
@@ -20,14 +20,11 @@ if [ "`npm view yarn@$version name`" = 'yarn' ]; then
   exit 0
 fi;
 
-# Grab the tarball so we can publish it as-is
-url=https://yarnpkg.com/downloads/$version/yarn-v$version.tar.gz
-tempfile=`mktemp -t 'yarn-release-XXXXXXXX.tar.gz'`
-curl --fail -L -o $tempfile $url
-
+# Determine if this is an RC or a stable release
+release_type=`curl --fail https://release.yarnpkg.com/release_type/$version`
```

Comment:
This endpoint simply returns "stable" or "rc" based off the version number passed in. In theory we could do the version comparison directly in this script, but it would still need to hit an external URL (`https://yarnpkg.com/latest-version`) to get the latest version and determine if this release is RC or stable, so I'd rather keep this simply and reuse the existing logic.